File: src/datastructure/stack/MyStack.java
package datastructure.stack;

import java.util.ArrayList;

public class MyStack <T>{
    ArrayList<T> stack = new ArrayList<>();
    public T pop(){
        if(!stack.isEmpty()){
            return  stack.get(stack.size() - 1);
        }
        return null;
    }

    public boolean isEmpty(){
        return stack.isEmpty();
    }

    public boolean push(T item){
        return stack.add(item);
    }
}

File: src/datastructure/stack/MyStack.kt
package datastructure.stack

class MyStack<T>(
    private val list: MutableList<T> = mutableListOf()
) {
    fun push(item: T): Boolean {
        return list.add(item)
    }
    fun pop(): T? {
        return if (list.isNotEmpty()) list.removeAt(list.size - 1) else null
    }
}

fun main() {
    val stack: MyStack<Int> = MyStack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    println(stack.pop())
    println(stack.pop())
    println(stack.pop())
}
